src/languages/graphql.js

```diff
@@ -0,0 +1,79 @@
+/*
+ Language: graphql
+ Category: scripting, protocols, web
+ Author: John Foster [jf990][], and others
+ Description: Highlight GraphQL queries with highlight.js.
+*/
+/** @type LanguageFn */
+export default function(hljs) {
+  const GQL_NAME = '[_A-Za-z][_0-9A-Za-z]*';
+  const GQL_PUNCTUATION = '[\\!\\(\\)\\:\\=\\[\\]\\{\\|\\}]{1}';
+  return {
+    name: "graphql",
+    aliases: [ "gql" ],
+    case_insensitive: true,
+    disableAutodetect: false,
+    keywords: {
+      keyword: [
+        "query",
+        "mutation",
+        "subscription|10",
+        "type",
+        "input",
+        "schema",
+        "directive",
+        "interface",
+        "union",
+        "scalar",
+        "fragment|10",
+        "enum",
+        "on"
+      ],
+      literal: "true false null"
+    },
+    contains: [
+      hljs.HASH_COMMENT_MODE,
+      hljs.QUOTE_STRING_MODE,
+      hljs.NUMBER_MODE,
+      {
+        className: "punctuation",
+        begin: "[.]{2}",
+        end: "\\.",
+        relevance: 0
+      },
+      {
+        className: "punctuation",
+        begin: GQL_PUNCTUATION,
```

Comment:
```suggestion
        match: GQL_PUNCTUATION,
```

Prefer match if there is no `end`.